scripts/release/release.js

```diff
@@ -33,22 +35,23 @@ async function run() {
   }
 
   const steps = [
-    require("./steps/validate-new-version.js"),
-    require("./steps/check-git-status.js"),
-    require("./steps/install-dependencies.js"),
-    require("./steps/run-tests.js"),
-    require("./steps/update-version.js"),
-    require("./steps/generate-bundles.js"),
-    require("./steps/update-changelog.js"),
-    require("./steps/push-to-git.js"),
-    require("./steps/publish-to-npm.js"),
-    require("./steps/bump-prettier.js"),
-    require("./steps/update-dependents-count.js"),
-    require("./steps/post-publish-steps.js"),
+    "./steps/validate-new-version.js",
+    "./steps/check-git-status.js",
+    "./steps/install-dependencies.js",
+    "./steps/run-tests.js",
+    "./steps/update-version.js",
+    "./steps/generate-bundles.js",
+    "./steps/update-changelog.js",
+    "./steps/push-to-git.js",
+    "./steps/publish-to-npm.js",
+    "./steps/bump-prettier.js",
+    "./steps/update-dependents-count.js",
+    "./steps/post-publish-steps.js",
   ];
 
   try {
-    for (const step of steps) {
+    for (const file of steps) {
+      const { default: step } = await import(file);
```

Comment:
What do you think about using `Promise.all` for importing all step modules?

```js
function importDefault(file) {
  return import(file).then((m) => m.default);
}
```

```js
const steps = await Promise.all([
  importDefault("./steps/validate-new-version.js"),
  importDefault("./steps/check-git-status.js"),
  importDefault("./steps/install-dependencies.js"),
  importDefault("./steps/run-tests.js"),
  importDefault("./steps/update-version.js"),
  importDefault("./steps/generate-bundles.js"),
  importDefault("./steps/update-changelog.js"),
  importDefault("./steps/push-to-git.js"),
  importDefault("./steps/publish-to-npm.js"),
  importDefault("./steps/bump-prettier.js"),
  importDefault("./steps/update-dependents-count.js"),
  importDefault("./steps/post-publish-steps.js"),
]);

try {
  for (const step of steps) {
    await step(params);
  }
} catch (error) {
  ...
}
```

I think it is readable and faster.

Reply:
https://github.com/prettier/prettier/pull/11229/commits/282cae5ba430385f8f980baca4b2a840da7ec579